pkg/antctl/transform/addressgroup/transform.go

```diff
@@ -18,23 +18,52 @@ import (
 	"io"
 	"reflect"
 
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/cmd/get"
+	"k8s.io/kubectl/pkg/scheme"
+
 	"antrea.io/antrea/pkg/antctl/transform"
 	"antrea.io/antrea/pkg/antctl/transform/common"
 	cpv1beta "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
 )
 
+var flags = map[string]string{
+	"name":              "{.metadata.name}",
+	"creationTimestamp": "{.metadata.creationTimestamp}",
+	"uid":               "{.metadata.uid}",
+	"resourceVersion":   "{.metadata.resourceVersion}",
+}
+
 type Response struct {
 	Name  string               `json:"name" yaml:"name"`
 	Pods  []common.GroupMember `json:"pods,omitempty"`
 	Nodes []common.GroupMember `json:"nodes,omitempty"`
 }
 
 func listTransform(l interface{}, opts map[string]string) (interface{}, error) {
-	groups := l.(*cpv1beta.AddressGroupList)
-	result := []interface{}{}
-	for i := range groups.Items {
-		item := groups.Items[i]
-		o, _ := objectTransform(&item, opts)
+	groupsList := l.(*cpv1beta.AddressGroupList)
+	sortField := opts["sort-by"]
+	if sortField == "" {
+		sortField = "{.metadata.name}"
```

Comment:
I think the expected input of sort-by flag can be `.metadata.name`, you don't have to maintain a `flags` map which is unnecessary.

Reply:
@luolanzone 
This approach could be user friendly where in user can just type name instead of using json field like  {.metadata.name}, hence the mapping
name  - >  {.metadata.name}
creationTimestamp - > {.metadata.creationTimestamp}
uid ->              {.metadata.uid}

Reply:
It also makes it easier to differentiate between 1) standard sorting using default ordering on json fields (e.g., `.metadata.name`), and 2) special cases such as `effectivePriority` where we use a custom ordering on a set of fields.